src/net.cpp

```diff
@@ -2783,18 +2833,21 @@ std::vector<AddedNodeInfo> CConnman::GetAddedNodeInfo() const
     }
 
     for (const auto& addr : lAddresses) {
-        CService service(LookupNumeric(addr.m_added_node, GetDefaultPort(addr.m_added_node)));
+        CService service{MaybeFlipIPv6toCJDNS(LookupNumeric(addr.m_added_node, GetDefaultPort(addr.m_added_node)))};
```

Comment:
> MaybeFlipIPv6toCJDNS

Is there any way to better encapsulate this `MaybeFlipIPv6toCJDNS()` usage? This seems to be a frequent pattern (a number of additional calls were added in #27071), and it seems like something that would be better handled at a lower level, rather than having to continually remember to wrap various networking calls.

I guess this is also able to remain broken is various ways because we don't have an specific tests for this functionality?